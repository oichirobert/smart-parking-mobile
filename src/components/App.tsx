import React, { useEffect } from 'react';
import { useScreens } from 'react-native-screens';
import Amplify, { Analytics } from 'aws-amplify';
import { withAuthenticator } from 'aws-amplify-react-native';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import awsconfig from '../../aws-exports'
import Navigation from './Navigation';
import reducers from '../reducers'
import * as actions from '../actions';

const store = createStore(reducers, applyMiddleware(thunk));

Amplify.configure(awsconfig);
Analytics.configure({ disabled: true });
useScreens();

function App() {
  useEffect(() => {
    store.dispatch(actions.fetchUser());
  });

  return (
    <Provider store={store}>
      <Navigation />
    </Provider>
  );
}

export default withAuthenticator(App);
