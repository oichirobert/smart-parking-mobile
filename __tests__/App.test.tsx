import 'react-native';
import React from 'react';
import renderer, { act } from 'react-test-renderer';
import App from '../src/components/App';

jest.mock('react-native-maps', () => {
  const React = require.requireActual('react');
  const MapView = require.requireActual('react-native-maps');

  class MockCallout extends React.Component {
    render() {
      return React.createElement('Callout', this.props, this.props.children);
    }
  }

  class MockMarker extends React.Component {
    render() {
      return React.createElement('Marker', this.props, this.props.children);
    }
  }

  class MockMapView extends React.Component {
    render() {
      return React.createElement('MapView', this.props, this.props.children);
    }
  }

  MockCallout.propTypes = MapView.Callout.propTypes;
  MockMarker.propTypes = MapView.Marker.propTypes;
  MockMapView.propTypes = MapView.propTypes;
  MockMapView.Marker = MockMarker;
  MockMapView.Callout = MockCallout;
  return MockMapView;
});

describe('App', () => {
  it('renders correctly', async () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });
});
