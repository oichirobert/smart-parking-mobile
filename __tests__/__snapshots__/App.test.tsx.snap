// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-around",
      "paddingTop": 20,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "padding": 20,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 32,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#152939",
            "fontSize": 20,
            "fontWeight": "500",
          }
        }
      >
        Loading...
      </Text>
    </View>
  </View>
  <Text>
    Please Sign In / Sign Up
  </Text>
</RCTSafeAreaView>
`;
